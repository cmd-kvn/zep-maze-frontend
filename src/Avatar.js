import React from 'react';
import {emojify} from 'react-emojione';
import { UP, DOWN, LEFT, RIGHT } from './Controller.constants';

export function Avatar({ stance }) {
    switch(stance) {
        case UP: return ( <div> emojify( ':point_up_tone3:' ) </div> );
        case DOWN: return ( <div> emojify( ':point_down_tone3:' ) </div> );
        case RIGHT: return ( <div> emojify(':point_right_tone3:') </div> );
        case LEFT: return ( <div> emojify( ':point_left_tone3:' ) </div> );
        default: return ( <div> emojify( ':levitate_tone3:' ) </div>);
    }
}