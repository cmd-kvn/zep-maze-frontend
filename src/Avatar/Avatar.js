import React from 'react';
import  { emojify } from 'react-emojione';
import { connect } from 'react-redux';
import styles from './Avatar.css'

export default function Avatar() {
    
    return ( <div className={styles.avatar}> {emojify(':wink:')} </div>);
}
