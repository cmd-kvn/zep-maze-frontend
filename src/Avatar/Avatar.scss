

:local(.avatar){
    // margin-left: -16px;
    // margin-top: -16px;
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    padding: auto;
    display: flex;   
    justify-content: center;
    align-items: center;
}