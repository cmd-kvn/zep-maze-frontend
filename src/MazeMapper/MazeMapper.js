import React, { Component } from 'react';
import { Tile } from '../Tile/Tile';
import { connect } from 'react-redux';
import styles from './MazeMapper.css';

export class MazeMapper extends Component{

    constructor(props){
        super(props);
    }    
    
    // componentDidMount() {
    //     fetch('/api/mazes')
    //         .then(res => res.json())
    //         .then(mazes => mazes[0].matrix = this.state.matrix);
    // }

    render() {

        const { position, maze } = this.props;

        // const matrix = maze.matrix;
        
        const xVal = position[0] * 44;
        const yVal = position[1] * 44;
        
        const transform = `transform: translate( ${ xVal }px, ${ yVal }px )`;

        return (
            <table className={styles.maze} style={{transform}}>
                <tbody>
                    {maze.matrix.map( ( row, index ) => {
                        return (
                            <tr key={index}>
                            {row.map( ( tile, index ) => {
                                return <Tile key={index} value={tile}/>
                            })}
                            </tr>
                        )
                    })}
                </tbody>
            </table>
    )}
}

export default connect(state => ({
    position: state.position,
    gamestate: state.gamestate,
    maze: state.maze,
    loading: state.loading,
    error: state.error
}))(MazeMapper);