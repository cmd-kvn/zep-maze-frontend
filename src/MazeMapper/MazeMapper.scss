:local(.maze){

    border-collapse: collapse;
    width: max-content;

}