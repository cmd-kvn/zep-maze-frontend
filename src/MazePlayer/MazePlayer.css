:local(.player) {
  margin: 20px auto; }
  :local(.player) .viewport {
    width: 144px;
    height: 144px;
    overflow: hidden;
    position: relative;
    margin: auto; }
