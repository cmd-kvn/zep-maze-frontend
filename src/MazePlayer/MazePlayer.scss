$size: 144px;


:local(.player){
    margin: 20px auto;
    .viewport{
        width: $size;
        height: $size;
        overflow: hidden;
        position: relative;
        margin: auto;
    }
}