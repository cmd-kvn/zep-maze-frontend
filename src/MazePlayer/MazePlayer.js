import React from 'react';
import styles from './MazePlayer.css';
import MazeMapper from '../MazeMapper/MazeMapper';
import Avatar from '../Avatar/Avatar';
import ControllerContainer from '../Controller/ControllerContainer';

export default function MazePlayer({ maze, loading, error, position, gamestate }) {

    if(loading || !maze) return <div>Loading Mazes...</div>;

    return (
        <div className={styles.player}>
            <div className="viewport">
                <MazeMapper maze={maze} position={position}/>
                <Avatar />
            </div>
            <ControllerContainer maze={maze} position={position} />
        </div>
    )
}