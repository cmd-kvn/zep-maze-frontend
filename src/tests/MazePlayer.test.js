import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MazePlayer from '../MazePlayer/MazePlayer';
import Avatar from '../Avatar/Avatar';
import MazeMapper from '../MazeMapper/MazeMapper';
import { WALL, FLOOR, START, END } from '../Tile/Tile.constants';

const maze = {
    score: [],
    matrix: [
        [ WALL, WALL, WALL, WALL, WALL ],
        [ WALL, START, WALL, END, WALL ],
        [ WALL, FLOOR, WALL, FLOOR, WALL ],
        [ WALL, FLOOR, FLOOR, FLOOR, WALL ],
        [ WALL, WALL, WALL, WALL, WALL ],
    ]
}

describe('MazePlayer component', () => {
    it('renders a  loading div when no maze passed', () => {
        const wrapper = shallow(<MazePlayer 
            maze={ null  } 
            loading={ false }
            error={ null }
            position={[ 0, 0 ]}
            gamestate={'ACTIVE'}
        />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders a div with a maze and an avatar when child components passed', () => {
        const wrapper = shallow(
            <MazePlayer 
                MazeMapper={MazeMapper}  
                Avatar={Avatar}/>
            );
        expect(toJSON(wrapper)).toMatchSnapshot();
    })
})