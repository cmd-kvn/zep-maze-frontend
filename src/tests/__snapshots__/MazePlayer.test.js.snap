// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MazePlayer component renders a  loading div when no maze passed 1`] = `
<div>
  Loading Mazes...
</div>
`;

exports[`MazePlayer component renders a div with a maze and an avatar when child components passed 1`] = `
<div>
  Loading Mazes...
</div>
`;
